### PR TITLE
Shepherdly config

### PR DESCRIPTION
**Pull request by mgreene**
Shepherdly now supports `yml` config at the repo level as well as custom mitigations.

Another notable change is the support of custom mitigations. Per @def-'s [request](https://shepherdly.slack.com/archives/C051BG4BFL7/p1704815224163999), I've added a custom mitigation that links out to running nightly tests. 

Everything else in the file is a copy of how the repo settings were configured to date.

### Motivation
Shepherdly is moving towards injectable configuration and this change brings the `materialize` repo up to parity.

### CODE REVIEW

**Review by def- — APPROVED** [review]
Perfect, thank you!

**Comment by def-** [comment]
I don't have permission to push into your branch:
```
ERROR: Permission to shepherdlyinc/materialize.git denied to def-.
```

To get the linting green you can add this (or give me push permissions):
```diff
diff --git a/.config/shepherdly.yml b/.config/shepherdly.yml
index ff94997c22..b08b501d21 100644
--- a/.config/shepherdly.yml
+++ b/.config/shepherdly.yml
@@ -1,3 +1,12 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
 # Docs: https://help.shepherdly.io/en/articles/8818618-shepherdly-repo-config
 configurations:
   - name: scoring_comment_enabled
@@ -31,4 +40,4 @@ mitigations:
     custom: true
     label: Run Nightly Tests
     label_url: https://trigger-ci.dev.materialize.com/?pr={{pr_number}}
-default_system_mitigations: false
\ No newline at end of file
+default_system_mitigations: false
```

File: .config/shepherdly.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # A value of 100 effectively makes this always optional
+    score_threshold: 100
+  - name: observability
+    score_threshold: 75
+  - name: run_nightly_tests
+    score_threshold: 80
+    custom: true
+    label: Run Nightly Tests
+    label_url: https://trigger-ci.dev.materialize.com/?pr={{pr_number}}
+default_system_mitigations: false
```
Comment (def-) [inline .config/shepherdly.yml:44]: Still no newline at the end of file (thus not a valid file in the POSIX sense)

**Review by def- — CHANGES_REQUESTED** [review]

File: .config/shepherdly.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (def-) [inline .config/shepherdly.yml:45]: Now it's two newlines, but doesn't matter ;)

**Review by def- — APPROVED** [review]

File: .config/shepherdly.yml (hunk 2 above)
Comment (def-) [reply .config/shepherdly.yml:45]: Actually does matter: `.config/shepherdly.yml:45: new blank line at EOF.`

**Comment by nrainer-materialize** [comment]
This PR contains unrelated commits, I assume a rebase went wrong.

**Pull request closed without merging**